esp/src/src-react/components/Results.tsx

```diff
@@ -0,0 +1,146 @@
+    wuid
+}) => {
+
+    const [grid, setGrid] = React.useState<any>(undefined);
+    const [selection, setSelection] = React.useState([]);
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+    const [results] = useWorkunitResults(wuid);
+
+    //  Command Bar  ---
+    const buttons: ICommandBarItemProps[] = [
+        {
+            key: "refresh", text: nlsHPCC.Refresh, iconProps: { iconName: "Refresh" },
+            onClick: () => refreshTable()
+        },
+        { key: "divider_1", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+    ];
+
+    const rightButtons: ICommandBarItemProps[] = [
+        {
+            key: "copy", text: nlsHPCC.CopyWUIDs, disabled: !uiState.hasSelection || !navigator?.clipboard?.writeText, iconOnly: true, iconProps: { iconName: "Copy" },
+            onClick: () => {
+                const wuids = selection.map(s => s.Wuid);
+                navigator?.clipboard?.writeText(wuids.join("\n"));
+            }
+        },
+        {
+            key: "download", text: nlsHPCC.DownloadToCSV, disabled: !uiState.hasSelection, iconOnly: true, iconProps: { iconName: "Download" },
+            onClick: () => {
+                Utility.downloadToCSV(grid, selection.map(row => ([row.Protected, row.Wuid, row.Owner, row.Jobname, row.Cluster, row.RoxieCluster, row.State, row.TotalClusterTime])), "workunits.csv");
+            }
+        }
+    ];
+
+    //  Grid ---
+    const gridStore = useConst(new Observable(new MyMemory()));
+    gridStore.alphanumSort["Name"] = true;
+    gridStore.alphanumSort["Value"] = true;
+
+    const gridSort = useConst([{ attribute: "Wuid", "descending": true }]);
+    const gridColumns = useConst({
+        col1: selector({
+            width: 27,
+            selectorType: "checkbox"
+        }),
+        Name: {
+            label: nlsHPCC.Name, width: 180, sortable: true,
+            formatter: function (Name, idx) {
+                return "<a href='#' onClick='return false;' class='dgrid-row-url'>" + Name + "</a>";
+            }
+        },
+        FileName: {
+            label: nlsHPCC.FileName, sortable: true,
+            formatter: function (FileName, idx) {
+                return "<a href='#' onClick='return false;' class='dgrid-row-url2'>" + FileName + "</a>";
+            }
+        },
+        Value: {
+            label: nlsHPCC.Value,
+            width: 180,
+            sortable: true
+        },
+        ResultViews: {
+            label: nlsHPCC.Views, sortable: true,
+            formatter: function (ResultViews, idx) {
+                let retVal = "";
+                ResultViews.forEach((item, idx) => {
+                    retVal += "<a href='#' onClick='return false;' viewName=" + encodeURIComponent(item) + " class='dgrid-row-url3'>" + item + "</a>&nbsp;";
+                });
+                return retVal;
+            }
+        }
+    });
+
+    const refreshTable = (clearSelection = false) => {
```

Comment:
@GordonSmith did you see this warning? Should this be wrapped in `useCallback`?

Reply:
No I hadn't, will fix...

Reply:
Fixed.